src/bootstrap/build/flags.rs

```diff
@@ -0,0 +1,102 @@
+// Copyright 2015 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+use std::fs;
+use std::path::PathBuf;
+use std::process;
+use std::slice;
+
+use getopts::Options;
+
+pub struct Flags {
+    pub verbose: bool,
+    pub stage: Option<u32>,
+    pub build_target: String,
+    pub host: Filter,
+    pub target: Filter,
+    pub build: Filter,
+    pub test: Filter,
+    pub config: Option<PathBuf>,
+    pub src: Option<PathBuf>,
+    pub jobs: Option<u32>,
+    pub args: Vec<String>,
+}
+
+pub struct Filter {
+    values: Vec<String>,
+}
+
+impl Flags {
+    pub fn parse(args: &[String]) -> Flags {
+        let mut opts = Options::new();
+        opts.optflag("v", "verbose", "use verbose output");
+        opts.optopt("", "config", "TOML configuration file for build", "FILE");
+        opts.optmulti("", "host", "host to build", "HOST");
+        opts.reqopt("", "build-target", "target of the stage0 build compiler", "BUILD");
+        opts.optmulti("", "target", "target to build", "TARGET");
+        opts.optmulti("", "build", "target to build", "BUILD");
```

Comment:
"target" is "target to build" and "build" is "target to build".

Can you say that "host" is the "host triple to build", "targe" the "target triple", and rename "build" and its description to something more clear? I know "build" sounds right in context, but it means something completely different in the existing rust/clang/gcc build systems.